x-pack/plugins/data_visualizer/public/application/common/components/results_links/results_links.tsx

```diff
@@ -12,10 +12,23 @@ import { EuiFlexGroup, EuiFlexItem, EuiCard, EuiIcon } from '@elastic/eui';
 import { TimeRange, RefreshInterval } from '@kbn/data-plugin/public';
 import { FindFileStructureResponse } from '@kbn/file-upload-plugin/common';
 import type { FileUploadPluginStart } from '@kbn/file-upload-plugin/public';
+import { flatten } from 'lodash';
+import { LinkCardProps } from '../link_card/link_card';
 import { useDataVisualizerKibana } from '../../../kibana_context';
+import { isDefined } from '../../util/is_defined';
 
 type LinkType = 'file' | 'index';
 
+export interface AsyncLinkCardParams {
+  dataViewId: string;
+  dataViewTitle?: string;
+  globalState?: any;
+}
+
+export type AsyncLinkCards = Array<
+  (params: AsyncLinkCardParams) => Promise<ResultLink | ResultLink[] | undefined>
```

Comment:
It would probably simplify the code if we only ever returned an array of `ResultLink` rather than having to deal with single links.

Reply:
Updated here https://github.com/elastic/kibana/pull/131342/commits/846bd87fbe77a488d47e1a9ee7d2d56a17011e74